src/requested_path.rs

```diff
@@ -13,15 +14,27 @@ fn decode_percents(string: &&str) -> String {
     percent_decode(string.as_bytes()).decode_utf8().unwrap().into_owned()
 }
 
+fn normalize_path(path: &Path) -> PathBuf {
+    path.components().fold(PathBuf::new(), |mut result, p| {
+        match p {
+            Component::Normal(x) => {
+                result.push(x);
+                result
+            }
+            Component::ParentDir => {
+                result.pop();
+                result
+            },
+            _ => result
+        }
+    })
+}
+
 impl RequestedPath {
     pub fn new<P: AsRef<Path>>(root_path: P, request: &Request) -> RequestedPath {
+        let decoded_req_path = PathBuf::from_iter(request.url.path().iter().map(decode_percents));
```

Comment:
Ah, I meant:

Suppose we have the following file systems:

- /web-server-root
- /web-server-root/index.html
- /web-server-too/directory-i-do-not-have-permission-to-access/

... and then serve static files from "/web-server-root".

In this case, the request, "directory-i-do-not-have-permission-to-access/xxx/../../index.html" should be successful, I thought. It looks Python3's builtin web server allows it, at least.

I do not have a strong opinion either. :)